File: javascript/grade-school/grade-school.js
var School = function() {
  this.students = {};
};

School.prototype.roster = function() {
  return this.students;
};

School.prototype.add = function(student, grade) {
  this.students[grade] = this.students[grade] || [];
  this.students[grade].push(student);
  this.students[grade].sort();
};

School.prototype.grade = function(grade) {
  return this.students[grade] || [];
};

module.exports = School;

File: ruby/grade-school/grade_school.rb
class School
  def initialize
    @enrollment = Hash.new { |h, k| h[k] = [] }
  end

  def students(grade)
    enrollment[grade]
  end

  def add(name, grade)
    enrollment[grade].push(name).sort!
  end

  def students_by_grade
    enrollment.keys.sort.map { |i| { grade: i, students: enrollment[i] } }
  end

  private

  attr_reader :enrollment
end

module BookKeeping
  VERSION = 3
end
